pkg/connector/handlematrix.go

```diff
@@ -0,0 +1,160 @@
+package connector
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"maunium.net/go/mautrix/bridgev2"
+	"maunium.net/go/mautrix/bridgev2/database"
+	"maunium.net/go/mautrix/bridgev2/networkid"
+	"maunium.net/go/mautrix/event"
+
+	"go.mau.fi/mautrix-twitter/pkg/twittermeow/data/payload"
+)
+
+var (
+	MSG_TYPE_TO_MEDIA_TYPE = map[event.MessageType]payload.MediaType{
+		event.MsgVideo: payload.MEDIA_TYPE_VIDEO_MP4,
+		event.MsgImage: payload.MEDIA_TYPE_IMAGE_JPEG,
+	}
+	MSG_TYPE_TO_MEDIA_CATEGORY = map[event.MessageType]payload.MediaCategory{
+		event.MsgVideo: payload.MEDIA_CATEGORY_DM_VIDEO,
+		event.MsgImage: payload.MEDIA_CATEGORY_DM_IMAGE,
+	}
+)
+
+var (
+	_ bridgev2.ReactionHandlingNetworkAPI    = (*TwitterClient)(nil)
+	_ bridgev2.ReadReceiptHandlingNetworkAPI = (*TwitterClient)(nil)
+	_ bridgev2.EditHandlingNetworkAPI        = (*TwitterClient)(nil)
+)
+
+func (tc *TwitterClient) HandleMatrixMessage(ctx context.Context, msg *bridgev2.MatrixMessage) (message *bridgev2.MatrixMessageResponse, err error) {
+	conversationId := string(msg.Portal.ID)
+	sendDMPayload := &payload.SendDirectMessagePayload{
+		ConversationID:    conversationId,
+		IncludeCards:      1,
+		IncludeQuoteCount: true,
+		RecipientIds:      false,
+		DmUsers:           false,
+	}
+
+	if msg.ReplyTo != nil {
+		sendDMPayload.ReplyToDmID = string(msg.ReplyTo.ID)
+	}
+
+	content := msg.Content
+	if content.FileName != "" && content.Body != content.FileName {
+		sendDMPayload.Text = content.Body
+	}
```

Comment:
This block would make more sense inside the video/image `case`

Reply:
`content.Body` may also be the filename when `content.FileName` is empty

Reply:
> `content.Body` may also be the filename when `content.FileName` is empty

yup but that's never set here - I will add that condition anyways